pennylane/decomposition/resources.py

```diff
@@ -133,8 +135,12 @@ def __repr__(self):
         return f"{self.op_type.__name__}({params})" if self.params else self.op_type.__name__
 
 
-def _make_hashable(d) -> tuple:
-    return tuple((k, _make_hashable(v)) for k, v in d.items()) if isinstance(d, dict) else d
+def _make_hashable(d):
```

Comment:
To what extent this private helper will be in use? I'm just wondering should we consider more property for `hashable` here e.g. whether two identical `dict` will be successfully hashed to the same `tuple`, given that in the first branch we manually implemented some part of hashing whose *item order* is not clearly preserved (or Python secrectly did that already? If so then ignore this comment). I suppose an inplace `sort` with keyvalues might clear the worry here, but performance could be issue.
If this method is just restrictvely merely used in few places where we don't care about more property then also this comment can be marked as resolved immediately.

Reply:
Good point. It should definitely be sorted.